cpp/recognition/planar/HopcroftTarjan.cpp

```diff
@@ -0,0 +1,358 @@
+                                        std::vector<NetworKit::node> &lowpt2,
+                                        std::vector<NetworKit::node> &parent) {
+        dfsnum[v] = dfs_count++;
+        lowpt2[v] = lowpt[v] = dfsnum[v];
+        reached[v] = true;
+        for (auto w : g.neighborRange(v)) {
+            if (!reached[w]) {
+                Add.push_back(NetworKit::Edge(v, w));
+                parent[w] = v;
+                dfs_in_reorder(g, w, Add, dfs_count, reached, dfsnum, lowpt, lowpt2,
+                               parent);
+                lowpt[v] = std::min(lowpt[v], lowpt[w]);
+            } else {
+                lowpt[v] = std::min(lowpt[v], dfsnum[w]);
+                if (dfsnum[w] < dfsnum[v] && parent[v] != w) {
+                    Add.push_back({v, w});
+                }
+            }
+        }
+        for (auto w : g.neighborRange(v)) {
+            if (parent[w] == v) {
+                if (lowpt[w] != lowpt[v]) {
+                    lowpt2[v] = std::min(lowpt2[v], lowpt[w]);
+                }
+                lowpt2[v] = std::min(lowpt2[v], lowpt2[w]);
+            } else {
+                if (lowpt[v] != dfsnum[w]) {
+                    lowpt2[v] = std::min(lowpt2[v], dfsnum[w]);
+                }
+            }
+        }
+    }
+
+    void HopcroftTarjan::reorder(NetworKit::Graph &g,
+                                 std::vector<NetworKit::node> &parent,
+                                 std::vector<NetworKit::node> &dfsnum,
+                                 std::vector<NetworKit::Edge> &order) {
+        NetworKit::node v = 0;
+        std::vector<bool> reached(N);
+        int dfs_count = 0;
+        std::vector<NetworKit::Edge> Add;
+        std::vector<NetworKit::node> lowpt(N), lowpt2(N);
+        dfs_in_reorder(g, v, Add, dfs_count, reached, dfsnum, lowpt, lowpt2,
+                       parent);
+        std::vector<NetworKit::Edge> buckets[N * 2 + 5];
+        for (auto [u, v] : Add) {
+            int cost = ((dfsnum[v] < dfsnum[u])
+                        ? 2 * dfsnum[v]
+                        : ((lowpt2[v] >= dfsnum[u]) ? 2 * lowpt[v]
+                                                    : 2 * lowpt[v] + 1));
+            buckets[cost].push_back(NetworKit::Edge(u, v));
+        }
+        g = NetworKit::Graph(N, true, true);
+        order = std::vector<NetworKit::Edge>(Add.size());
+        int id = 0;
+        for (int i = 0; i <= 2 * N; i++) {
+            for (auto [u, v] : buckets[i]) {
+                g.addEdge(u, v, id);
+                order[id] = NetworKit::Edge(u, v);
+                id++;
+            }
+        }
+    }
+
+    bool HopcroftTarjan::strong_planar(NetworKit::Graph &g, NetworKit::Edge e,
+                                       std::list<int> &Att, std::vector<int> &alpha,
+                                       std::vector<NetworKit::node> &dfsnum,
+                                       std::vector<NetworKit::node> &parent) {
```

Comment:
Maybe https://networkit.github.io/dev-docs/cpp_api/classNetworKit_1_1StronglyConnectedComponents.html might be useful